models/channel_helpers.go

```diff
@@ -287,9 +286,13 @@ func ChangeChannel(q *reform.Querier, channelID string, params *ChangeChannelPar
 
 // RemoveChannel removes notification channel with specified id.
 func RemoveChannel(q *reform.Querier, id string) error {
+	if _, err := FindChannelByID(q, id); err != nil {
+		return err
+	}
+
 	err := q.Delete(&Channel{ID: id})
 	if err != nil {
-		return errors.Wrap(err, "failed to delete notifications channel")
+		return errors.Wrap(err, "failed to delete notification channel")
```

Comment:
`q.Delete` has this in docs `Method returns ErrNoRows if no rows were deleted.`. So the same behavior can be achieved without  TX. Something like that:

```
func RemoveChannel(q *reform.Querier, id string) error {
	if id == "" {
		return status.Error(codes.InvalidArgument, "Empty Channel ID.")
	}
	
	err := q.Delete(&Channel{ID: id})
	if err != nil {
		if err == reform.ErrNoRows {
			return status.Errorf(codes.NotFound, "Channel with ID %q not found.", id)
		}
		return errors.Wrap(err, "failed to delete notification channel")
	}
	return nil
}
```

No?

Reply:
Sure, but then we duplicate those checks between Find, Change and Remove

Reply:
Hm, ok. PMMs DB is not a hot place, nothing bad happened because of an extra transaction)

Reply:
Exactly. Plus PostgreSQL's MVCC is a great help for us there – unrelated transactions are running in full speed